src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -53,25 +57,40 @@
 import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 
+import static com.amazon.opendistroforelasticsearch.security.DefaultObjectMapper.getOrDefault;
+
 /**
  * This class represents all configurations for compliance.
  * DLS/FLS uses this configuration for filtering and anonymizing fields.
  * Audit Logger uses this configuration to post compliance audit logs.
  */
+@JsonAutoDetect(getterVisibility = JsonAutoDetect.Visibility.NONE)
 public class ComplianceConfig {
 
     private static final Logger log = LogManager.getLogger(ComplianceConfig.class);
+    public static final ComplianceConfig DEFAULT = ComplianceConfig.from(Settings.EMPTY);
     private static final int CACHE_SIZE = 1000;
     private static final String INTERNAL_ELASTICSEARCH = "internal_elasticsearch";
 
     private final boolean logExternalConfig;
```

Comment:
Shouldn't this be @JsonProperty(value="external_config")? Any reason why it is not mentioned here and added here - https://github.com/opendistro-for-elasticsearch/security/pull/542/files#diff-0e04f575b10578faf6c21e321ea023b3R293 ?

Same for following lines - https://github.com/opendistro-for-elasticsearch/security/pull/542/files#diff-0e04f575b10578faf6c21e321ea023b3R80-R82?

Reply:
Because other properties are used internally within the class only. They dont have associated getter. 
Jackon prefers getter binding and if not available uses property binding.

Reply:
To be more exact, it is not Jackson that prefers getter. By annotating getters, we tell Jackson to use getters during serialization and use `@JsonCreator` during de-serialization. For those properties that don't have getters, there is no other options.